### DISCUSSION TITLE
Prevent dragging from target handle

### DISCUSSION THREAD

**Discussion by atruttmann** [xyflow/xyflow#2169] [Q&A]
I have a target handle that should accept connections from other handles. I want to prevent drag events when I click and drag on the target handle. 

My use case: it's a large invisible handle covering my node. I want to still be able to drag my actual node without starting a connection. I can't use `isConnectable={false}` on my target handle since then it won't accept connections.

Is there a workaround for this or documentation I'm missing? Thanks!

**Comment by bcakmakoglu** [comment]
Have you tried using the `noDragClassName` prop?
It defaults to `nodrag` I believe, so you can just add the class `nodrag` to your handle and that should be it.

**Reply by atruttmann** [reply]
Thanks for the response! Your answer helped me find the correct course of action (or at least one that works). I'll write my solution here in case anyone is looking to do something similar. The use case I'm trying to recreate is how Figma's prototyping mode works in terms of handles, where the whole layer is a target handle.

Since my node and handle are the same, on click + drag of the node I wanted it to be moved around like a normal node. The problem I was having was the handle wanted to start making a new edge. So instead of adding the `"nodrag"` class, I removed it to allow the node be dragged.

Within my custom node, I added this:
```
const targetHandleRef = useRef();
const removeNoDragClass = () => {
    if (targetHandleRef.current && targetHandleRef.current.classList.contains("nodrag")) {
        targetHandleRef.current.classList.remove("nodrag"); // remove any time class is added
    }
};
useMutationObserver(targetHandleRef, removeNoDragClass);
useEffect(() => {
    targetHandleRef.current.classList.remove("nodrag"); // remove on mount
}, []);
```

Sometimes the `nodrag` class is added to the handle on click, so you will also add some sort of mutation observer to remove it every time it happens. You also want to remove it on mount.
